scripts/calculateKpi/protocols/tetherV0/index.test.ts

```diff
@@ -0,0 +1,265 @@
+  const startTimestamp = new Date('2024-01-01T00:00:00Z')
+  const endTimestampExclusive = new Date('2024-01-31T23:59:59Z')
+
+  const defaultProps = {
+    address: testAddress,
+    startTimestamp,
+    endTimestampExclusive,
+  }
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    mockGetHyperSyncClient.mockReturnValue(mockClient)
+    mockGetBlockRange.mockResolvedValue({
+      startBlock: 1000,
+      endBlockExclusive: 2000,
+    })
+  })
+
+  describe('calculateKpi', () => {
+    it('should calculate KPI across all supported networks', async () => {
+      await calculateKpi(defaultProps)
+
+      // Verify getBlockRange was called for each network
+      expect(mockGetBlockRange).toHaveBeenCalledTimes(8)
+    })
+
+    it('should handle networks with no eligible transactions', async () => {
+      // Mock paginateQuery to return no transactions
+      mockPaginateQuery.mockImplementation(async (_client, _query, onPage) => {
+        const mockResponse = makeQueryResponse([])
+        await onPage(mockResponse)
+      })
+
+      const result = await calculateKpi(defaultProps)
+
+      expect(result).toEqual({
+        kpi: 0,
+        metadata: {
+          [NetworkId['ethereum-mainnet']]: 0,
+          [NetworkId['avalanche-mainnet']]: 0,
+          [NetworkId['celo-mainnet']]: 0,
+          [NetworkId['unichain-mainnet']]: 0,
+          [NetworkId['ink-mainnet']]: 0,
+          [NetworkId['op-mainnet']]: 0,
+          [NetworkId['arbitrum-one']]: 0,
+          [NetworkId['berachain-mainnet']]: 0,
+        },
+      })
+    })
+
+    it('should throw errors from getBlockRange', async () => {
+      mockGetBlockRange.mockRejectedValue(new Error('Block range error'))
+
+      await expect(calculateKpi(defaultProps)).rejects.toThrow(
+        'Block range error',
+      )
+    })
+
+    it('should throw errors from paginateQuery', async () => {
+      mockPaginateQuery.mockRejectedValue(new Error('Query error'))
+
+      await expect(calculateKpi(defaultProps)).rejects.toThrow('Query error')
+    })
+  })
+
+  describe('getEligibleTxCount', () => {
+    const encodedValueAboveThreshold = ('0x' +
+      BigNumber(2).shiftedBy(6).toString(16).padStart(64, '0')) as Hex
+    const encodedValueBelowThreshold = ('0x' +
+      BigNumber(0.5).shiftedBy(6).toString(16).padStart(64, '0')) as Hex
+
+    it('should count transactions with transfer value above minimum threshold', async () => {
+      mockPaginateQuery.mockImplementation(async (_client, _query, onPage) => {
+        const mockResponse = makeQueryResponse([
+          {
+            data: encodedValueAboveThreshold,
+            topics: [
+              transferEventSigHash,
+              pad(testAddress, { size: 32 }),
+              pad('0x4567890123456789012345678901234567890123' as Address, {
+                size: 32,
+              }),
+              '0x0000000000000000000000000000000000000000000000000000000000000000',
+            ],
```

Comment:
supernit: we could maybe consider using `encodeEventTopics` and `encodeAbiParameters` from viem to make it more readable?

Something similar to what we did in https://github.com/divvi-xyz/divvi-relayer/blob/a55f8fc95a8b16547336e22207e4af0f23b324a2/src/services/divviRegistry.test.ts#L98-L113